src/events/google_cloud.js

```diff
@@ -79,6 +86,143 @@ function bigQueryWrapper(wrappedFunction) {
+                        break;
+                    case 'createTopic':
+                        responseMetadata.topic = arg2;
+                        break;
+                    case 'deleteTopic':
+                        if (config.reqOpts && config.reqOpts.topic) {
+                            responseMetadata.topic = utils.getLastSplittedItem(
+                                config.reqOpts.topic,
+                                '/'
+                            );
+                        }
+                        break;
+                    default:
+                        break;
+                    }
+                    eventInterface.finalizeEvent(pubsubEvent, startTime, err, responseMetadata);
+                    resolve();
+                    if (callback) {
+                        callback(err, arg2, ...arg3);
+                    }
+                };
+            });
+            tracer.addEvent(pubsubEvent, responsePromise);
+        } catch (err) {
+            tracer.addException(err);
+        }
+        return original.apply(this, [config, patchedCallback]);
+    };
+}
+
+/**
+ * Wrap pubsub request function.
+ * @param {Function} original pubsub request function.
+ * @returns {Function} The wrapped function
+ */
+function wrapPubSubPullFunction(original) {
+    return function internalPubSubPullFunction(request, options, callback) {
+        let clientPromiseRequest;
+        try {
+            let pubsubProjectId;
+            if (request && request.subscription) {
+                const subscriptionSplited = request.subscription.split('/');
+                if (subscriptionSplited.length > 1) {
+                    [, pubsubProjectId] = subscriptionSplited;
+                }
+            }
+            const { slsEvent: pubsubEvent, startTime } = eventInterface.initializeEvent(
+                GOOGLE_CLOUD_TYPES.pubsub.name, pubsubProjectId, 'Pull', GOOGLE_CLOUD_TYPES.pubsub.name
+            );
+            const patchedCallback = (err, res) => {
+                const responseMetadata = {};
+                if (res && res.receivedMessages) {
+                    const receivedMessages = res.receivedMessages.reduce((acc, current) => {
+                        acc.push({ messageId: current.message.messageId, message: `${current.message.data}` });
+                        return acc;
+                    }, []);
+                    responseMetadata.receivedMessages = receivedMessages;
+                }
+                eventInterface.finalizeEvent(pubsubEvent, startTime, err, responseMetadata);
+                if (callback) {
+                    callback(err, res);
+                }
+            };
+            // in case callback was given from the client.
+            if (callback) {
+                let patchedCallbackWithPromise = callback;
+                const promise = new Promise((resolve) => {
+                    patchedCallbackWithPromise = (err, res) => {
+                        resolve();
```

Comment:
not good enough, cause it should be *after* we update the event. you may need to split `patchedCallback`.

Reply:
fixed.